### PR TITLE
Fix biocro tests

### LINKED ISSUE #1320: Failing test: model2netcdf.BIOCRO
After #1312, model2netcdf.BIOCRO needs a result that contains AboveLitter and BelowLitter, but run.biocro doesn't yet return these, causing test.model2netcdf.R to stop with an error.

- [x] Update run.biocro.R to include litter in hourly results
- [x] Save new version of `tests/testthat/data/result.RData` with litter variables included
- [x] Uncomment in testthat.R, fix Travis errors

### PR DESCRIPTION
**Pull request by infotroph**
Fixes #1320, #1328.

## Description
* Updated run.biocro and model2netcdf to retain litter variables in output
* Saved new version of sample biocro results. **CAUTION**: My change to result.RData adds dummy values (AboveLitter = BelowLitter = 0) rather than real output. If it's important for result.RData ought to contain a genuine representative BioCro result, this should be changed.
* Uncommented all tests
* Suppressed warning by not trying to unlink files that don't exist
* Suppressed password prompt/warning by changing run host to localhost

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: models/biocro/R/run.biocro.R [hunk 1]
```diff
@@ -138,6 +139,8 @@ run.biocro <- function(lat, lon, metpath, soil.nc = NULL, config = config, coppi
   daily.results <- hourly.results[, list(Stem = max(Stem), 
                                          Leaf = max(Leaf),
                                          Root = max(Root), 
+                                         AboveLitter = max(AboveLitter),
```
Comment (dlebauer) [inline models/biocro/R/run.biocro.R]: if this is hourly, why would it be max?

File: models/biocro/R/model2netcdf.BIOCRO.R [hunk 2]
```diff
@@ -77,6 +77,7 @@ model2netcdf.BIOCRO <- function(result, genus = NULL, outdir, lat = -9999, lon =
       TotLivBiom = k * (Leaf + Root + Stem + Rhizome + Grain), 
       RootBiom = k * Root, 
       StemBiom = k * Stem, 
+      Yield = Stem,
```
Comment (dlebauer) [inline models/biocro/R/model2netcdf.BIOCRO.R]: Perhaps we should leave 'yield' out completely, as the components of yield vary by crop (e.g. maize --> grain, willow --> stem, biomass grass -->  stem + leaf

File: models/biocro/tests/testthat/test.model2netcdf.BIOCRO.R [hunk 3]
```diff
@@ -1,39 +1,37 @@
 context("check output from model2netcdf.BIOCRO")
 
-# library(data.table)
-# library(PEcAn.utils)
-# library(PEcAn.settings)
+library(data.table)
+library(PEcAn.utils)
+library(PEcAn.settings)
```
Comment (dlebauer) [inline models/biocro/tests/testthat/test.model2netcdf.BIOCRO.R]: is this necessary now that read.settings is called using `package::function()` notation?

File: models/biocro/tests/testthat/test.write.configs.BIOCRO.R [hunk 4]
```diff
@@ -1,14 +1,13 @@
...
-# settings <- PEcAn.settings::prepare.settings(settings)
+settings.xml <- file.path("data", "pecan.biocro.xml")
+settings <- PEcAn.settings::read.settings(settings.xml)
+settings <- PEcAn.settings::prepare.settings(settings)
 
-# samples <- list(biocro.saof = (data.frame(Vcmax = c(31.9, 42.4, 57), cuticular_cond = c(1800, 
-#   4380, 10700), leaf_respiration_rate_m2 = c(1, 1.9, 3.6), stomatal_slope.BB = c(2.7, 
-#   3.3, 3.9), row.names = c("15.866", "50", "84.134"))))
+samples <- list(biocro.saof = (data.frame(Vcmax = c(31.9, 42.4, 57), cuticular_cond = c(1800, 
+  4380, 10700), leaf_respiration_rate_m2 = c(1, 1.9, 3.6), stomatal_slope.BB = c(2.7,
```
Comment (dlebauer) [inline models/biocro/tests/testthat/test.write.configs.BIOCRO.R]: odd locations for line breaks

File: models/biocro/R/run.biocro.R (hunk 1 above)
Comment (infotroph) [inline models/biocro/R/run.biocro.R]: I wondered that too -- presumably to guarantee each var has length 1? BioCro will happily produce multiple lines per hour if given sub-hourly input, but maybe that's a pathology that should be caught further upstream.

File: models/biocro/tests/testthat/test.model2netcdf.BIOCRO.R (hunk 3 above)
Comment (infotroph) [inline models/biocro/tests/testthat/test.model2netcdf.BIOCRO.R]: Nope. Removed.

**Comment by infotroph** [comment]
@dlebauer Fixed all comments except "remove use of `max()`...", clarification needed for that one: Were you reading `max(Stem)` as an unnecessary wrapper around values already of length 1? The section you highlighted is computing daily values *from* hourly ones, so it's necessary to use a summary function of some kind on them. Do you want to switch to `mean()` instead of `max()`

File: models/biocro/R/run.biocro.R (hunk 1 above)
Comment (dlebauer) [inline models/biocro/R/run.biocro.R]: Sorry I read the code too quickly and thought this was hourly not daily. This looks ok. do h2o quantities have time in the denominator (fluxes) or would sum work for sub hourly data?

If easiest route is to require hourly time steps then please add an error if either met input or model output are not hourly.

**Comment by dlebauer** [comment]
Keep max for daily and yearly output.

File: models/biocro/R/run.biocro.R (hunk 1 above)
Comment (infotroph) [inline models/biocro/R/run.biocro.R]: Canopy transpiration has time in the denominator, precip doesn't, and evaporation it depends which BioCro function's docs I look at :p 

I''ve opened a separate ticket (#1338) to make run.biocro check for hourly input.

**Pull request merged**